notes/stv/20220304-test-deploy-validate-checksums.txt

```diff
@@ -0,0 +1,271 @@
+                        "filepath": "https://raw.githubusercontent.com/wfau/aglais-testing/main/notebooks/Library_validation.json",
+                        "totaltime": 60,
+                        "results": {}
+                }
+
+
+        ]
+}
+
+
+	
+
+# -----------------------------------------------------
+# Run Quick test
+#[root@ansibler]
+
+    num_users=1
+    concurrent=False
+    test_level=quick
+
+    # Restart Zeppelin
+    time \
+        /deployments/hadoop-yarn/bin/restart-zeppelin.sh
+
+    time \
+        /deployments/hadoop-yarn/bin/run-tests.sh \
+            "${cloudname:?}" \
+            "${configname:?}" \
+            "${test_level:?}"  \
+	     ${concurrent:?}  \
+	     ${num_users:?}  \
+        | tee /tmp/run-tests-quick.log
+
+
+> Done
+
+TASK [Run benchmarker] **************************************************************************************************************************************************************************************
+changed: [localhost] => {"changed": true, "cmd": "python3 /tmp/run-test.py | tee /tmp/test-result.json", "delta": "0:02:11.675368", "end": "2022-03-04 13:52:53.840886", "rc": 0, "start": "2022-03-04 13:50:42.165518", "stderr": "", "stderr_lines": [], "stdout": "Test started [Single User]\nTest completed! (131.48 seconds)\n------------ Test Completion: [SLOW] ------------ Test Output: [INVALID] ------------\n[{'SetUp': {'totaltime': '46.97', 'status': 'SLOW', 'msg': '', 'valid': 'TRUE'}, 'Mean_proper_motions_over_the_sky': {'totaltime': '53.51', 'status': 'SUCCESS', 'msg': 'Expected/Actual output missmatch of cell #0! ', 'valid': 'FALSE'}, 'Source_counts_over_the_sky.json': {'totaltime': '21.32', 'status': 'SUCCESS', 'msg': 'Expected/Actual output missmatch of cell #1! ', 'valid': 'FALSE'}, 'Library_Validation.json': {'totaltime': '9.67', 'status': 'SUCCESS', 'msg': '', 'valid': 'TRUE'}}]", "stdout_lines": ["Test started [Single User]", "Test completed! (131.48 seconds)", "------------ Test Completion: [SLOW] ------------ Test Output: [INVALID] ------------", "[{'SetUp': {'totaltime': '46.97', 'status': 'SLOW', 'msg': '', 'valid': 'TRUE'}, 'Mean_proper_motions_over_the_sky': {'totaltime': '53.51', 'status': 'SUCCESS', 'msg': 'Expected/Actual output missmatch of cell #0! ', 'valid': 'FALSE'}, 'Source_counts_over_the_sky.json': {'totaltime': '21.32', 'status': 'SUCCESS', 'msg': 'Expected/Actual output missmatch of cell #1! ', 'valid': 'FALSE'}, 'Library_Validation.json': {'totaltime': '9.67', 'status': 'SUCCESS', 'msg': '', 'valid': 'TRUE'}}]"]}
+
+
+
+
+# Formatted results:
+
+------------ Test Completion: [SLOW] ------------ Test Output: [INVALID] ------------
+
+
+[{
+	'SetUp': {
+		'totaltime': '54.00',
+		'status': 'SLOW',
+		'msg': '',
+		'valid': 'TRUE'
+	},
+	'Mean_proper_motions_over_the_sky': {
+		'totaltime': '67.90',
+		'status': 'SLOW',
+		'msg': 'Expected/Actual output missmatch of cell #0! ',
+		'valid': 'FALSE'
```

Comment:
This is what we need to avoid.

We need one value that indicates if the test passes or fails. If we are just checking the `status`, then `SLOW` suggests that the test passed but was slower than expected. Having to check a combination of both `status` and `valid` to determine if the test passed or not is a recipe for confusion.

Other suggestions aside (#649) the overall test result should be represented in a single value, so if the output validation fails, then `status` should be set to `FAIL`. Completing the test run but producing the wrong results should not be considered a success.

Reply:
I've made some modifications to the benchmarker to match your suggestions at (#649) and some notes testing these changes

Reply:
:+1: